persistent/Database/Persist/SqlBackend.hs

```diff
@@ -0,0 +1,150 @@
+-- | This module contains types and information necessary for a SQL database.
+-- Database support libraries, like @persistent-postgresql@, will be responsible
+-- for constructing these values.
+module Database.Persist.SqlBackend
+    ( -- * The type and construction
+      SqlBackend
+    , mkSqlBackend
+    , MkSqlBackendArgs(..)
```

Comment:
Breaking changes to the `MkSqlBackendArgs` type (representing *required* parameters) will be a breaking change to the library.